File: RotateImage.java
import java.util.*;

class RorateImage {
    public void rotate(int[][] matrix) {
        int[][] d=matrix;
        int n2=matrix[0].length-1;
        for(int i=0; i<d.length; i++) {
            int n1=matrix.length-1;
            for(int j=0; i<d[i].length; j++) {
                matrix[n1][n2]=d[i][j];
                n1--;
            }
            n2--;
        }
    }
}

File: neetcode/RotateImage.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        vector<vector<int>> result(matrix.size(), vector<int>(matrix.size()));
        for(int i=0; i<matrix.size(); ++i) {
            for(int j=0; j<matrix[i].size(); ++j) {
                result[j][matrix.size()-1-i]=matrix[i][j];
            }
        }
        matrix=result;
    }
};
